build/builder.go

```diff
@@ -568,13 +568,25 @@ func (b *Builder) Finish() error {
 
 		delete(toDelete, final)
 
-		b.shardLog("upsert", final)
+		b.shardLog("upsert", final, b.opts.RepositoryDescription.Name)
 	}
 	b.finishedShards = map[string]string{}
 
 	for p := range toDelete {
+		if zoekt.TombstonesEnabled(filepath.Dir(p)) {
+			// Don't delete compound shards, set tombstones instead.
+			if strings.HasPrefix(filepath.Base(p), "compound-") {
+				if strings.HasSuffix(p, ".zoekt") {
```

Comment:
all this indentation is quite intense. Is there a way to flatten a bit?

Reply:
yep, there is :-)